dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -194,6 +198,26 @@ private static Validator createValidator() {
         return validatorFactory.getValidator();
     }
 
+    /**
+     * Decide whether a gitReference is excluded, given a workflow/service's filters
+     * @param gitRefPath Path.of(gitReference) for glob matching with PathMatcher
+     * @param filters Filters specified for a workflow/service in .dockstore.yml
+     * @return
+     */
+    public static boolean filterGitReference(final Path gitRefPath, final List<String> filters) {
+        return filters.isEmpty() || filters.stream().anyMatch(filter -> {
+            String matcherString;
+            // Use regex if filter string is surrounded by /
+            if (filter.matches("^\\/.*\\/$")) {
+                matcherString = "regex:^refs\\/(heads\\/|tags\\/|)" + filter.substring(1, filter.length() - 1);
```

Comment:
What happens if the filter is not a valid regex, which will inevitably happen? Does the code on line 216 thrown an exception? Took a brief look at the code, and I think it does throw an exception:



Some thoughts:

1. Let it throw an exception, and we don't process the .dockstore.yml
2. Catch the exception, and don't process the .dockstore.yml
3. Catch the exception, and treat it as a filter that didn't pass, move on to next filter.
4. Validate the filters when loading the .dockstore.yml and treat it as an invalid .dockstore.yml to start with.

Reply:
Great point, something like `/[/` would throw an error.

Probably best to handle the error here https://github.com/dockstore/dockstore/pull/3733/files#diff-d792d91142c26ee5507333d7d3db9acfR337